src/linkplay/endpoint.py

```diff
@@ -38,3 +46,38 @@ async def json_request(self, command: str) -> dict[str, str]:
 
     def __str__(self) -> str:
         return self._endpoint
+
+
+class LinkPlayTcpUartEndpoint(LinkPlayEndpoint):
+    """Represents a LinkPlay TCPUART API endpoint."""
+
+    def __init__(self, *, endpoint: str):
```

Comment:
I wonder if we should pass the `StreamReader` and `StreamWriter` from the constructor. That way the responsibility of opening/closing them falls on the one creating the `LinkPlayTcpUartEndpoint`. They can also then be instantiated with a `with` statement which closes them for you.